apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,220 @@
+	//
+	// +optional
+	TLS *BackendTLSConfig `json:"tls,omitempty" protobuf:"bytes,2,opt,name=tls"`
+}
+
+// BackendRef identifies an API object within a known namespace that defaults
+// group to core and resource to services if unspecified.
+type BackendRef struct {
+	// Group is the group of the referent.  Omitting the value or specifying
+	// the empty string indicates the core API group.  For example, use the
+	// following to specify a service:
+	//
+	// fooRef:
+	//   resource: services
+	//   name: myservice
+	//
+	// Otherwise, if the core API group is not desired, specify the desired
+	// group:
+	//
+	// fooRef:
+	//   group: acme.io
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +optional
+	// +kubebuilder:default=core
+	Group string `json:"group" protobuf:"bytes,1,opt,name=group"`
+
+	// Resource is the API resource name of the referent. Omitting the value
+	// or specifying the empty string indicates the services resource. For example,
+	// use the following to specify a services resource:
+	//
+	// fooRef:
+	//   name: myservice
+	//
+	// Otherwise, if the services resource is not desired, specify the desired
+	// group:
+	//
+	// fooRef:
+	//   group: acme.io
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +optional
+	// +kubebuilder:default=services
+	Resource string `json:"resource" protobuf:"bytes,2,opt,name=resource"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
+
+	// Port is the port of the referent. If unspecified, this policy applies to
+	// all ports on the backend.
+	// +optional
+	Port *int32 `json:"port,omitempty" protobuf:"varint,4,opt,name=port"`
+}
+
+// BackendTLSConfig describes TLS configuration for a backend.
+type BackendTLSConfig struct {
+	// ClientCertificateRef is a reference to a TLS client certificate that may
+	// be used to connect to these backends. If an entry in this list omits or
+	// specifies the empty string for both the group and the resource, the
+	// resource defaults to "secrets". An implementation may support other
+	// resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
+	//
+	// Support: Extended
+	//
+	// +optional
+	ClientCertificateRef *CertificateObjectReference `json:"clientCertificateRef,omitempty" protobuf:"bytes,1,rep,name=clientCertificateRef"`
+
+	// ClientCertificateKeyRef is a reference to a TLS client certificate key
+	// that may be used to connect to these backends. If an entry in this list
+	// omits or specifies the empty string for both the group and the resource,
+	// the resource defaults to "secrets". An implementation may support other
+	// resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
+	//
+	// Support: Extended
+	//
+	// +optional
+	ClientCertificateKeyRef *CertificateObjectReference `json:"clientCertificateKeyRef,omitempty" protobuf:"bytes,2,rep,name=clientCertificateKeyRef"`
```

Comment:
Having a separate object Reference for certificate and key seems wrong.

Client certificate work similar to Server certificates, meaning, a user can put the key and certificate in the same secret (or whatever CRD) and then refer both of them at the same time.

We should follow the same pattern we have in TLSConfig here.

Reply:
Kubernetes secrets have a tls secret type which can be reused here.

Reply:
Updated, thanks!